### ISSUE TITLE
AnimationTree 'Flick' in one shot node conncted with a transition node if there is state change

### ISSUE THREAD

**Issue by nagasawamaki** [godotengine/godot#23665]
**Godot version:**
3.1 `a16d9c6`

**OS/device including version:**
Windows 7

**Issue description:**
There is a flick in one shot node if there is state change in transition node.
You can see below gif.
![2018-11-12_18-11-34](https://user-images.githubusercontent.com/24987226/48340786-72d01800-e6a6-11e8-9bb5-213d1d2bb283.gif)

**Steps to reproduce:**
import a DAE model, with animations.
Create a scene based on the model
Create a `AnimationTree` node
In the `AnimationTree` node, create `AnimationNodeBlendTree`
In the `AnimationNodeBlendTree`, create three `AnimationNode`, one `Transition` and one `OneShot`
Connect one `AnimationNode` to the `OneShot`'s `in` and connect the `Transition` to the `OneShot`'s `shot`
Connect the other two `AnimationNode` as inputs to the `Transition`

Then you can try to click the `OneShot` on, and try to change the state of the `Transition` and try to click the `OneShot` on again and you will see the flick.

**Minimal reproduction project:**
[test_project.zip](https://github.com/godotengine/godot/files/2571441/test_project.zip)

In this project, you can use key board key UP and Down to change the one shot animation and see the flick effect

**Comment by pouleyKetchoupp** [comment]
I've encountered a similar issue when combining `Transition` nodes together.

From what I've seen the problem generally happens when a `Transition` node is used as an input for another multiple input node, like `Transition` or `OneShot`.

If you try to switch the current value for the transition while it's inactive, next time it's active it will try to blend between the last value and the new one, causing the flickering.

It doesn't seem like the expected behavior, as in this case, you would rather want the `Transition` to act as a switch for the output node, and not blend from whatever state it was in last time it was used. But I'm not sure how to fix it in the animation system.

As a workaround, I was able to fix this issue by script by setting the property `prev_current` to be -1 when setting `current` in order to force the `Transition` to switch to the new animation instead of blending.

@nagasawamaki I've tested my workaround in your repro project and it does the trick:
```
func _physics_process(delta):
	if (Input.is_action_just_pressed("ui_down")):
		$AnimationTree["parameters/Transition/prev_current"] = -1
		$AnimationTree["parameters/Transition/current"] = 0
		$AnimationTree["parameters/OneShot/active"] = true
		pass
	
	if (Input.is_action_just_pressed("ui_up")):
		$AnimationTree["parameters/Transition/prev_current"] = -1
		$AnimationTree["parameters/Transition/current"] = 1
		$AnimationTree["parameters/OneShot/active"] = true
		pass
	pass
```

**Comment by KoBeWi** [comment]
Still valid in 7c4791e6d

**Comment by blurrred** [comment]
Thank you for reporting this, as well as a workaround. I was having the same issue and didn't know what was causing it. Your workaround fixed the problem.
This is with Godot 3.3.

**Comment by theludovyc** [comment]
Hello :) !

I reproduce same in 2d. I'm in 3.4.3.

I made a minimal exemple project. Just open AnimBlendTree. Choose Idle0 in Idles. Activate OneShot. Now choose Idle1 in Idles.

[Issue_23665.zip](https://github.com/godotengine/godot/files/8541138/Issue_23665.zip)

**Comment by theludovyc** [comment]
Hello :) !

I reproduce same in 2d. I'm in 3.4.4.

I made a minimal exemple project. Just open AnimBlendTree. Choose Idle0 in Idles. Activate OneShot. Now choose Idle1 in Idles.

[Issue_23665.zip](https://github.com/godotengine/godot/files/8541188/Issue_23665.zip)

**Comment by Hiiamwilliam** [comment]
Hey, I checked both MRPs in Godot 3.5.stable. The original issue might have been fixed.

I cannot reproduce OP's issue, I see no flick like the gif. The only abrupt movement happens when I activate the OneShot while it's still "shooting", but that's expected I think.

The second MRP seems to be a different issue, related to https://github.com/godotengine/godot/issues/39496. It seems the `AnimationNodeBlendTree` incorrectly keeps the OneShot active.
In this MRP, firing the OneShot when it isn't contributing to the `AnimationNodeOutput` makes it remain active. Then, switching the Transition so that the OneShot does contribute makes it play its `shot` animation, since it is active. If you deactivate the OneShot before switching the Transition, no "flick" happens.
